File: tests/perf/array_for_in.py
#!/usr/bin/env python
""" Tests iteration over array using for-in loop. """

def run():
    """ Body of the test """

    loops = 300
    size = 10000
    elements = list(range(size))
    total = 0

    loop_idx = 0
    while loop_idx < loops:
        loop_idx += 1

        for elem in elements:
            total += elem

    print(total)

run()

File: tests/perf/array_for_in.js
const loops = 300;
const size  = 10000;
const a     = [];
let   total = 0;

a.length = size;
for (let i = 0; i < size; i++) {
    a[i] = i;
}

for (let l = 0; l < loops; l += 1) {

    for (let elem in a) {
        total += a[elem];
    }
}

print(total);
